zipline/data/data_portal.py

```diff
@@ -190,6 +190,8 @@ def get_current_price_data(self, asset, column, dt=None):
         column_to_use = self.column_lookup[column]
 
         if self.data_frequency == "daily":
+            dt = pd.Timestamp(dt.date(), tz='utc')
```

Comment:
I'm still hesitant about fixing the input here, as opposed to having the caller (period/tracker) pass `datetime.date` for which it requesting the data, but it is definitely ok while everything is coming together and perhaps more long term than that.